logstash-core/src/main/java/org/logstash/ackedqueue/Queue.java

```diff
@@ -946,7 +945,7 @@ private void maybeSignalReadDemand(boolean forceSignal) {
         // if we're not forcing, and if the current read demand has
         // neither been met nor expired, this method becomes a no-op.
         if (!forceSignal && Objects.nonNull(readDemand)) {
-            if (unreadCount < readDemand.elementsNeeded && System.currentTimeMillis() < readDemand.deadlineMillis) {
+            if (!readDemand.isSatisfiable(unreadCount)) {
```

Comment:
With my other comment that replaces and inverts `ReadDemand#isSatisfiable` with `ReadDemand#isDeferrable`:

```suggestion
            if (readDemand.isDeferrable(unreadCount)) {
```